src/set.coffee

```diff
@@ -27,6 +27,7 @@ install = (_, jasmine) ->
   context = @
 
   globalPatches =
+    __autoIncrement: -1
```

Comment:
I had set it to -1 because in my original commit the id was 0 but autoincrement was 1 and the || between them was setting the id to 1. 

I changed it to 0 and it works fine with the reworked

```
id = jasmine?.getEnv()?.currentSpec?.id ? globalPatches.__autoIncrement++
```